server/base/src/main/java/org/apache/accumulo/server/util/SystemPropUtil.java

```diff
@@ -68,6 +69,52 @@ public static void setSystemProperty(ServerContext context, String property, Str
     context.getPropStore().putAll(SystemPropKey.of(context), Map.of(property, value));
   }
 
+  public static void setSystemProperties(ServerContext context, Map<String,String> propertiesMap) {
+    Map<String,String> checkedProperties = new HashMap<>();
+
+    for (Map.Entry<String,String> entry : propertiesMap.entrySet()) {
+      final var original = entry.getKey();
+      var property = entry.getKey();
+      var value = entry.getValue();
+
+      // Retrieve the replacement name for this property, if there is one.
+      // Do this before we check if the name is a valid zookeeper name.
+      property = DeprecatedPropertyUtil.getReplacementName(property,
+          (log, replacement) -> log
+              .warn("{} was deprecated and will be removed in a future release;"
+                  + " setting its replacement {} instead", original, replacement));
+
+      if (!Property.isValidZooPropertyKey(property)) {
+        IllegalArgumentException iae =
+            new IllegalArgumentException("Zookeeper property is not mutable: " + property);
+        log.debug("Attempted to set zookeeper property.  It is not mutable", iae);
+        throw iae;
+      }
+
+      // Find the property taking prefix into account
+      Property foundProp = null;
+      for (Property prop : Property.values()) {
+        if (prop.getType() == PropertyType.PREFIX && property.startsWith(prop.getKey())
+            || prop.getKey().equals(property)) {
+          foundProp = prop;
+          break;
+        }
+      }
+
+      if ((foundProp == null || (foundProp.getType() != PropertyType.PREFIX
+          && !foundProp.getType().isValidFormat(value)))) {
+        IllegalArgumentException iae = new IllegalArgumentException(
+            "Ignoring property " + property + " it is either null or in an invalid format");
+        log.debug("Attempted to set zookeeper property.  Value is either null or invalid", iae);
+        throw iae;
+      }
+
+      checkedProperties.put(property, value);
+    }
+
+    context.getPropStore().putAll(SystemPropKey.of(context), checkedProperties);
```

Comment:
This change may require updating the PropStore to replace the properties with these. Currently, `ZooPropStore.putAll` does a mutate of the existing properties to try to merge them into the existing properties. It does this by calling VersionedProperties.addOrUpdate. We probably need a new method on VersionedProperties, very similar to addOrUpdate, but doesn't initialize the new properties from the old ones. And then, that should be the method used by ZooPropStore in a new replaceAll (or setAll) method instead of calling putAll.

Reply:
See other comment concerning addOrUpdate instead of a replaceAll for set.  replaceAll may be appropriate as a second, explicit method, but I think the ability to add or modify a subset should be available.